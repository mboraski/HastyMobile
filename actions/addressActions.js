export const SAVE_ADDRESS = 'save_address';

export const saveAddress = address => dispatch => {
    dispatch({ type: SAVE_ADDRESS, payload: address });
};
