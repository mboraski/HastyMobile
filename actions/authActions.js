import { AsyncStorage } from 'react-native';
import { Facebook } from 'expo';

const ACTIONS = {
    FACEBOOK_LOGIN_SUCCESS: 'facebook_login_success',
    FACEBOOK_LOGIN_FAIL: 'facebook_login_fail'
};

/** Helper Functions **/
const doFacebookLogin = async dispatch => {
    try {
        const response = await Facebook.logInWithReadPermissionsAsync('1873998396207588', {
            permissions: ['public_profile', 'email', 'user_friends']
        });
        console.log('fb auth response ', response);
        const { type, token } = response;
        if (type === 'cancel') {
            return dispatch({ type: ACTIONS.FACEBOOK_LOGIN_FAIL });
        }
        await AsyncStorage.setItem('fb_token', token);
        dispatch({ type: ACTIONS.FACEBOOK_LOGIN_SUCCESS, payload: token });
    } catch (error) {
        console.log('doFacebookLogin catch: ', error);
    }
};
/** Helper Functions End **/

const facebookLogin = () => async dispatch => {
    const token = await AsyncStorage.getItem('fb_token');

    if (token) {
        console.log('fb_token', token);
        dispatch({ type: ACTIONS.FACEBOOK_LOGIN_SUCCESS, payload: token });
    } else {
        doFacebookLogin(dispatch);
    }
};

const ACTION_CREATORS = {
    facebookLogin
};

export default { ...ACTIONS, ...ACTION_CREATORS };
