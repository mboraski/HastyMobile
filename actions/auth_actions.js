import { AsyncStorage } from 'react-native';
import { Facebook } from 'expo';

import {
  FACEBOOK_LOGIN_SUCCESS,
  FACEBOOK_LOGIN_FAIL
} from './types';

// How to use AsyncStorage:
// AsyncStorage.setItem('fb_token', token);
// AsyncStorage.getItem('fb_token');
// AsyncStorage.removeItem('fb_token');

// bc we are returning everything in here, we can remove the return statement
export const facebookLogin = () => async dispatch => {
  let token = await AsyncStorage.getItem('fb_token');

  if (token) {
    // Dispatch an action saying FB login is done
    dispatch({ type: FACEBOOK_LOGIN_SUCCESS, payload: token });
  } else {
    // Start up FB Login process
    doFacebookLogin(dispatch); // we are passing dispatch in here
  }
};

const doFacebookLogin = async dispatch => {
  // remember redux thunk is allowing use to call actions at some later time here
  let { type, token } = await Facebook.logInWithReadPermissionsAsync('1873998396207588', {
    permissions: ['public_profile', 'email', 'user_friends']
  });

  if (type === 'cancel') {
    return dispatch({ type: FACEBOOK_LOGIN_FAIL });
  }

  await AsyncStorage.setItem('fb_token', token);
  dispatch({ type: FACEBOOK_LOGIN_SUCCESS, payload: token });
};
