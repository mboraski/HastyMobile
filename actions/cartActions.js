export const ADD_TO_CART = 'add_to_cart';

export const addToCart = (product) => dispatch => {
    dispatch({ type: ADD_TO_CART, payload: product });
};
