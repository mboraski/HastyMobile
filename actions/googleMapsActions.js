import googleMapsClient from '../api/googleMaps';

const KEY = 'AIzaSyDqrzisPO0B76kk7M7PYGemAgaHEvr7uBA';

export const MAPS_PLACES_AUTOCOMPLETE_REQUEST = 'google_maps_places_autocomplete_request';
export const MAPS_PLACES_AUTOCOMPLETE_SUCCESS = 'google_maps_places_autocomplete_success';
export const MAPS_PLACES_AUTOCOMPLETE_FAIL = 'google_maps_places_autocomplete_fail';

export const placesAutocomplete = input => async dispatch => {
    try {
        dispatch({
            type: MAPS_PLACES_AUTOCOMPLETE_REQUEST,
            payload: input
        });
        const res = await googleMapsClient.placesAutoComplete({
            params: {
                input,
                key: KEY
            }
        });
        if (res.data.error_message) {
            dispatch({
                type: MAPS_PLACES_AUTOCOMPLETE_FAIL,
                error: new Error(res.data.error_message)
            });
        } else {
            dispatch({
                type: MAPS_PLACES_AUTOCOMPLETE_SUCCESS,
                payload: res.data
            });
        }
    } catch (error) {
        dispatch({
            type: MAPS_PLACES_AUTOCOMPLETE_FAIL,
            error
        });
    }
};
