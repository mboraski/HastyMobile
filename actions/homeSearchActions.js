export const HOME_SEARCH_CHANGE = 'home_search_change';

export const homeSearchChange = (text) => dispatch => {
    dispatch({
        type: HOME_SEARCH_CHANGE,
        payload: {
            searchText: text,
        }
    });
};
