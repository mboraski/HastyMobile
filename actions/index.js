// export * from './auth_actions';
// export * from './job_actions';
export * from './cartActions';
