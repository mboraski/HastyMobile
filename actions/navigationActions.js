export const TOGGLE_MENU_OPEN = 'nav_toggle_menu_open';
export const TOGGLE_MENU_CLOSE = 'nav_toggle_menu_close';

export const openToggle = () => dispatch => {
    dispatch({ 
        type: TOGGLE_MENU_OPEN,
    });
};

export const closeToggle = () => dispatch => {
    dispatch({ 
        type: TOGGLE_MENU_CLOSE,
    });
};
