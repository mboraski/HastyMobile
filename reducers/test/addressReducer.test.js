import { saveAddress } from '../../actions/addressActions';
import reducer, { initialState } from '../addressReducer';

describe('addressReducer', () => {
    it('handles saveAddress action', () => {
        expect(reducer(initialState, saveAddress('address'))).toMatchSnapshot();
    });
});
