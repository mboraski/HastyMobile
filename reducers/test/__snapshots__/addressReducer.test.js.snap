// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addressReducer handles saveAddress action 1`] = `
Object {
  "error": null,
  "pending": false,
  "predictions": Array [],
  "saved": Array [
    "address",
  ],
}
`;
